src/main/java/org/alfresco/repo/management/JmxDumpUtil.java

```diff
@@ -164,9 +169,136 @@ private static void printMBeanInfo(MBeanServerConnection connection, ObjectName
                 attributes.put(OS_NAME, updateOSNameAttributeForLinux(osName));
             }
         }
+        if (objectName.getCanonicalName().equals("java.lang:type=Runtime"))
+        {
+            String[] commandInputs = (String[]) attributes.get(INPUT_ARGUMENTS);
+            if(commandInputs != null)
+            {
+                attributes.put(INPUT_ARGUMENTS, cleanPasswordsFromInputArguments(commandInputs,REDACTED_INPUTS));
+            }
+        }
         tabulate(JmxDumpUtil.NAME_HEADER, JmxDumpUtil.VALUE_HEADER, attributes, out, 0);
     }
 
+    /**
+     * Replaces strings with JmxDumpUtil.PROTECTED_VALUE, 
+     * if any of the string that contains a string from redactedInputs.
+     * 
+     * @see #cleanPasswordFromInputArgument
+     * @param commandInputs one or more strings of input arguments
+     * @param redactedInputs one or more strings, that end input arguments that are to be redacted
+     * @return commandInputs with any arguments ending in redactedInputs with redacted values
+     */
+    static String[] cleanPasswordsFromInputArguments(String[] commandInputs, String[] redactedInputs)
+    {
+        List<String> cleanInputs = new ArrayList<String>();
+        for (String input : commandInputs) 
+        {
+            input = cleanPasswordFromInputArgument(input, redactedInputs);
+            cleanInputs.add(input);
+        }
+        
+        return cleanInputs.toArray(new String[commandInputs.length]);
+    }
+
+    /**
+     * Removes any characters the word/s provided in redactedInputs
+     * and replaces them with JmxDumpUtil.PROTECTED_VALUE
+     * <p>
+     * Example: 
+     * <p>
+     * Input:   -Ddb.password=alfresco
+     * <p>
+     * Output:  -Ddb.password=********
+     * </p>
+     * 
+     * @param input String
+     * @param redactedInputs String[]
+     * @return password redacted string if input matches a string in redactedInputs, an un-altered string will be returned if it does not match.
+     */
+    static String cleanPasswordFromInputArgument(String input, String[] redactedInputs)
+    {
+        //Selects the whole string, if one of the redactedInputs are present. It will have two capture groups, group 1 is the proceeding token ie. "password=" and group 2 will be all characters following the = (the vaule/password).
+        String regex = createPasswordFindRegexString(redactedInputs); 

```

Comment:
This regex can be computed just once in `cleanPasswordsFromInputArguments`, since the `redactedInputs` list doesn't change between calls.

Reply:
The regex is now computed just once, in `cleanPasswordsFromInputArguments`.